framework/util/json_util.h

```diff
@@ -22,20 +22,168 @@
 */
 
 /// @file API-agnostic options and helpers for outputting JSON used by all
-/// the JSON-generating components such as the Consumers in.
-/// @note This currently looks overkill for one constant but there is a fuller
-/// version incoming from an in-progress PR.
+/// the JSON-generating components such as the Consumers that gfxrecon-convert
+/// is based on.
 
 #ifndef GFXRECON_UTIL_JSON_UTIL_H
 #define GFXRECON_UTIL_JSON_UTIL_H
 
 #include "util/defines.h"
+#include "util/to_string.h"
+#include "format/format.h"
+
+#include "nlohmann/json.hpp"
 
 GFXRECON_BEGIN_NAMESPACE(gfxrecon)
 GFXRECON_BEGIN_NAMESPACE(util)
 
 static constexpr int kJsonIndentWidth = 2;
 
+/// This selects between outputting an array of objects (JSON) or individual
+/// top-level objects separated by newlines (JSONL). A top-level component
+/// like Convert's main() needs to look at this but the FieldToJson functions
+/// ignore it. They build a nlohmann tree in memory.
+enum class JsonFormat : uint8_t
+{
+    JSON,
+    JSONL
+};
+
+/// Parameters potentially required in converting our datastructures to JSON.
+struct JsonOptions
```

Comment:
It looks like you thought it was important to pack this structure tightly (re: above making Json Format a uin8_t).  Can you comment here on why that was important?

Reply:
Not beyond it always being important and oft-neglected. Looking at the struct I see that by making the enum 8 bit, it and the following three bools will pack down nicely to 32 bits. Maybe that doesn't matter now  (it doesn't change the size of the struct on 64 bit platforms but does make it smaller on 32 bit: https://godbolt.org/z/4Wx44v8YE) but as the struct gets added to in the future it gives a whole extra three bytes to fill out (say with extra modal bools for future features) before the overall size of the struct increases or an int or float could be added unthinkingly by the next person to touch this code and the struct's size wouldn't increase now but would without the : uint8_t. There is no downside to making the enum as small as the minimum addressable unit of all known processors we anticipate targeting. It isn't like I used a bitfield to pack it into a single bit\*.

Keeping oft-accessed structs like this as small as possible is always a good thing. It reduces the chance of one crossing a page boundary and thus needing two TLB entries dedicated to it. It reduces the number of cachelines having to be kept around to hold it and thus contend with caching the main data the app is actually trying to process (think more of contending for ways in a four-way small L1 cache on a mid-tier ARM Android device than the relative size of it compared to the total MB of cache on a high end desktop and how that inevitably leads to thrashing cachelines between stack, this struct, and the data being read and written, and the only lever you have to pull to ameliorate is to make the struct as small as possible (and cacheline align it which I didn't do) and thus limit the number of lines thrashed each time it goes in and out of L1).

I think I've mentioned before that I take _'premature optimization is the root of all evil'_ with a huge grain of salt. An equally valid rule of thumb is _ignore optimization and suffer a performance death by a thousand cuts, no one of which will register in your profiles_. 

Give me a clear directive to delete the ` : uint8_t` and I will do so.

\* Nowadays when memory effects can trump code sequence opts for perf and power, bit-packing could actually be a win.